featurebyte/service/base_document.py

```diff
@@ -419,31 +439,34 @@ async def delete_document(
         int
             number of records deleted
         """
-        document = await self.get_document(
-            document_id=document_id,
-            exception_detail=exception_detail,
-            use_raw_query_filter=use_raw_query_filter,
-            disable_audit=self.should_disable_audit,
-            **kwargs,
-        )
+        with self.get_document_deletion_lock(
+            document_id=document_id, timeout=DOCUMENT_DELETION_LOCK_TIMEOUT
+        ):
```

Comment:
With this, the deletion request that waits for this lock would wait until the lock is release, and then fail immediately when the first deletion completes? Wonder if it makes more sense for such request to fail immediately.

Reply:
If i understood it correctly, do you mean modifying the lock so that it failed immediately? If that's the case, it makes sense to me. Will modify this accordingly.